pages/static/containers_js.go

```diff
@@ -696,12 +705,67 @@ function refreshStats() {
 			}
 			if (containerInfo.spec.has_network) {
 				startNetwork("network-selection", containerInfo);
+			}	
+			if (containerInfo.spec.has_custom_metrics) {
+				startCustomMetrics("custom-metrics-chart", containerInfo);
 			}
 		}
 			drawCharts(machineInfo, containerInfo);
 	});
 }
 
+function startCustomMetrics(elementId, containerInfo) {
+	var metrics = containerInfo.spec.custom_metrics;
+	var el=$("<div>");
+	for (i = 0; i<metrics.length; i++) {
+		divText = "<div id='"+elementId+"-"+metrics[i].name+"'></div>";
+		el.append($(divText));
+	}
+	el.append($("</div>"));
+	
+	$("#"+elementId).append(el);
+}
+
+function getCustomMetrics(rootDir, containerName, callback) {
+        $.getJSON(rootDir + "api/v2.0/appmetrics/" + containerName)
+        .done(function(data) {
+                callback(data);
+        })
+        .fail(function(jqhxr, textStatus, error) {
+	        callback([]);
+        });
+}
+
+function drawCustomMetrics(elementId, containerInfo, metricsInfo) {
+	if(metricsInfo.length == 0) {
+		return;
+	}
+
+	for (var containerName in metricsInfo) {
+		var container = metricsInfo[containerName];
+		var metricIndex = -1 ;
+		for (var metricName in container) {
+			metricIndex++;
+			var titles = ["Time", containerInfo.spec.custom_metrics[metricIndex].units];
```

Comment:
The order of stats and spec doesn't match. In my tests, the units gets mixed up.
Maps in go don't have a fixed order. I'd suggest changing the loop to iterate over the spec and locate spec.custom_metrics[index].name in metrics map returned by appmetrics.